### ISSUE TITLE
Travis CI build fails intermittently

### ISSUE THREAD

**Issue by timvandermeij** [mozilla/pdf.js#10732]
Sometimes the build fails because of a missing `maxCanvasPixels` property. Build https://travis-ci.org/mozilla/pdf.js/builds/519475880?utm_source=github_status&utm_medium=notification is an example with the following stack trace:

```
[22:18:21] Finished 'default_preferences-pre' after 7.48 s
[22:18:21] Starting '<anonymous>'...
[22:18:22] '<anonymous>' errored after 51 ms
[22:18:22] TypeError: Cannot read property 'maxCanvasPixels' of undefined
    at Object.<anonymous> (/home/travis/build/mozilla/pdf.js/build/default_preferences/lib/web/app_options.js:70:68)
    at Module._compile (internal/modules/cjs/loader.js:816:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10)
    at Module.load (internal/modules/cjs/loader.js:685:32)
    at Function.Module._load (internal/modules/cjs/loader.js:620:12)
    at Module.require (internal/modules/cjs/loader.js:723:19)
    at require (internal/modules/cjs/helpers.js:14:16)
    at /home/travis/build/mozilla/pdf.js/gulpfile.js:552:5
    at bound (domain.js:425:14)
    at runBound (domain.js:438:12)
[22:18:22] 'default_preferences' errored after 7.76 s
```

I think I have also seen it happening locally once, but I'm not entirely sure. Perhaps it's a race condition where the default preferences are not generated yet before the next task in executed?

**Comment by Snuffleupagus** [comment]
> Sometimes the build fails because of a missing `maxCanvasPixels` property.

Not sure that it matters much, but doesn't the error indicate that the problem rather is that the object (i.e. `viewerCompatibilityParams`) which is *holding* the property in question haven't been correctly initialized/loaded yet?

https://github.com/mozilla/pdf.js/blob/8dc2d21931a1ba73866d6df0a8b96df239bfd58f/web/app_options.js#L100

**Comment by Snuffleupagus** [comment]
> Perhaps it's a race condition where the default preferences are not generated yet before the next task in executed?

The problem might be that the `default_preferences` task runs *twice* during the setup phase, in the `lint` respectively the `lib` task, which I suppose could lead to intermittent issues with files not being found.

**Comment by timvandermeij** [comment]
Marking as a good beginner bug since only knowledge about Gulp should be enough to fix this. Basically we need to make sure that our tasks don't run more than once, keeping the existing dependencies.

**Comment by dhuang612** [comment]
Hi,
I would like to work on this!

**Comment by timvandermeij** [comment]
It's yours!

**Comment by dhuang612** [comment]
created PR #11411

**Comment by timvandermeij** [comment]
Fixed by #11413. I'll open a new issue for the Gulp 4 migration changes.
